lib/ld-eventsource/client.rb

```diff
@@ -204,47 +205,14 @@ def closed?
     
     def reset_http
       @http_client.close if !@http_client.nil?
-      @cxn = nil
-      @buffer = ""
-    end
-    
-    def read_lines
-      Enumerator.new do |gen|
-        loop do
-          line = read_line
-          break if line.nil?
-          gen.yield line
-        end
-      end
-    end
-    
-    def read_line
-      loop do
-        @lock.synchronize do
```

Comment:
This synchronization of access to `@buffer` using `@lock` has been removed in the refactored code - it was never necessary because the buffer is only ever accessed from one thread at a time.